graphite/graphite.go

```diff
@@ -42,23 +47,55 @@ func (r *Request) Query(host string) (Response, error) {
 	if r.End != nil {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
-	u := url.URL{
+	r.URL = url.URL{
 		Scheme:   "http",
 		Host:     host,
 		Path:     "/render/",
 		RawQuery: v.Encode(),
 	}
-	resp, err := DefaultClient.Get(u.String())
+	resp, err := DefaultClient.Get(r.URL.String())
 	if err != nil {
-		return Response{}, err
+		return nil, fmt.Errorf(requestErrFmt, r.URL.String(), "Get failed: "+err.Error())
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return Response{}, errors.New(resp.Status)
+		tb, err := readTraceback(resp)
+		if err != nil {
+			tb = &[]string{"<Could not read traceback: " + err.Error() + ">"}
+		}
+		return nil, fmt.Errorf(requestErrFmt, r.URL.String(), fmt.Sprintf("Get failed: %s\n%s", resp.Status, strings.Join(*tb, "\n")))
 	}
 	var series Response
 	err = json.NewDecoder(resp.Body).Decode(&series)
-	return series, err
+	if err != nil {
+		e := fmt.Errorf(requestErrFmt, r.URL.String(), "Parsing response failed: "+err.Error())
+		return series, e
+	}
+	return series, nil
+}
+
+func readTraceback(resp *http.Response) (*[]string, error) {
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	bodyLines := strings.Split(strings.TrimSpace(string(bodyBytes)), "\n")
+	tracebackLines := make([]string, 0, len(bodyLines))
```

Comment:
Let's leave out the cap here. I don't think it's winning us anything. just `var tracebackLines []string` should be enough.